src/RpcClient.ts

```diff
@@ -354,4 +382,111 @@ export class RpcClient {
       throw new Error(`Error in getSignaturesForAsset: ${error}`);
     }
   }
+  /**
+   * Mint a compressed NFT.
+   * @returns {Promise<MintCNFTResponse>}
+   * @throws {Error}
+   */
+  async mintCNFT(
+    request: MintCNFTRequest
+  ): Promise<MintCNFTResult | undefined> {
+    try {
+      const ownerKeypair = loadWallet(request.ownerKeypair);
+      function delay(ms: number) {
+        return new Promise((resolve) => setTimeout(resolve, ms));
+      }
+      const defaultMetadataArgs: MetadataArgs = {
+        ...request.metadataArgs,
+        primarySaleHappened: false,
```

Comment:
Are these fields optional? with this setup you will be overriding anything in metadata args with the values below.

Set the default above the spread (...request) for the user-provided values to override the defaults.